Options/Applicative.hs

```diff
@@ -14,6 +15,11 @@ module Options.Applicative (
   -- | The standard @Applicative@ module is re-exported here for convenience.
   module Control.Applicative,
 
+  -- | Re-export Data.Monoid's (<>) operator for convenience.
```

Comment:
This is a bit evil as then `optparse-applicative`'s exposed API depends on its transitive dependencies configuration.

Oh dear, I misread @HuwCampbell's original message. Yes, I agree, this is not good. Users shouldn't have to worry about the exposed interface for a given version of a package changing beneath their feet. I rescind my claim that this is a sensible way forward.

@HuwCampbell, I thought your suggestion here was to merely continue to unconditionally expose `Data.Monoid`. Really, I'm still not convinced that we shouldn't simply add a dependency on `semigroups` for GHC <8.9 and export its `(<>)`.